tests/sdk/nodejs/nodejs_test.go

```diff
@@ -887,6 +887,30 @@ func TestQuery(t *testing.T) {
 	integration.ProgramTest(t, &test)
 }
 
+// TestReadonlyMetadata tests the behavior of read-only metadata fields.
+func TestReadonlyMetadata(t *testing.T) {
+
+	test := baseOptions.With(integration.ProgramTestOptions{
+		Dir:   filepath.Join("metadata", "step1"),
+		Quick: true,
+		ExtraRuntimeValidation: func(t *testing.T, stackInfo integration.RuntimeValidationStackInfo) {
+			// Verify that the resourceVersion (a read-only property) is available as an output.
+			resourceVersion, ok := stackInfo.Outputs["resourceVersion"]
+			assert.Truef(t, ok, "missing expected output \"resourceVersion\"")
+			assert.NotEmptyf(t, resourceVersion, "resourceVersion is empty")
```

Comment:
Can we add another validation step to ensure the resourceVersion in the output is not `invalid-step1`.